### DISCUSSION TITLE
TypeScript type naming convention - Should `WP` be prefixed to types?

### DISCUSSION THREAD

**Discussion by johnhooks** [WordPress/gutenberg#48932] [Developer Experience]
I am working on adding types to `@wordpress/blocks` in PR #48604. A question that has come up, what is the convention regarding prefixing type names with `WP`? Module scoping prevents type names from clashing, and not all packages in Gutenberg prefix types with `WP`.

- The current JSDocs of `@wordpress/blocks` make reference to type declarations that do use the prefix and moving away from it would require updating documentation in other packages. Though since types would be new to `@wordpress/blocks`, no other packages depend on the `WP` prefixed names.

- The types exported by the DefinitelyTyped package [`@types/wordpress__blocks`](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/651e000cbf98277b0330bf14ea9b288a36387e75/types/wordpress__blocks) don't prefix the type names.

How should the types be named in `@wordpress/blocks`? More generally, is there a convention for type names and is it documented some where?

Vote here with a 👍 for _keeping_ the `WP` prefix and a 👎 for _getting rid of_ the `WP` prefix.

**Comment by johnhooks** [comment]
CC @ryanwelcher @dmsnell @adamziel @sirreal

**Comment by dmsnell** [comment]
I've long found the `WP` prefix odd and unnecessary. By nature of being in the WP packages they are already namespaced.

I'm quite happy with removing the prefixes, so this is my vote.

**Comment by johnhooks** [comment]
I would also vote to drop the prefix. (Though not entirely sure I'm a voting member lol)

**Comment by sirreal** [comment]
I don't see any benefit to having a WP prefix on types. I do see a drawback that the prefix adds noise making it harder to read the type name.

My vote would be to remove the prefix.

**Comment by dmsnell** [comment]
Added a suggestion to vote in the description at the top. cc: @johnhooks and @sirreal in case you want to pile on to the voting.
